File: ch4/exo3.pl
#!/usr/bin/env perl
use strict;
use warnings;
use utf8;

## print all of the keys and values in %ENV. 

my $lhs_length = 0;
foreach (sort keys %ENV) {
	if (length($_) > $lhs_length) {
		$lhs_length = length($_);
	}
}

foreach (sort keys %ENV) {
	printf("%*s => %s\n", $lhs_length, $_, $ENV{$_});
}

File: ch4/exo3.py
#!/usr/bin/env python3
import os 

## print environment variables
lhs_len = 0
for x in sorted(os.environ):
    if len(x) > lhs_len:
        lhs_len = len(x)

for x in sorted(os.environ):
    print("{1:{0}s} => {2}".format(lhs_len, x, os.environ[x]))
